railties/lib/rails/application_controller.rb

```diff
@@ -2,6 +2,8 @@ class Rails::ApplicationController < ActionController::Base # :nodoc:
   self.view_paths = File.expand_path('../templates', __FILE__)
   layout 'application'
 
+  before_action :set_locale
```

Comment:
Is this going to generic ApplicationController? It's an unrelated change to emails, although perhaps useful to discuss in it's own right.